### ISSUE TITLE
Installer

### ISSUE THREAD

**Issue by asentner** [opendevshop/devshop#577]
Has anyone actually had any success installing with the standalone installer?

I've tried on fresh servers with CentOS 7 and Ubuntu 18.0.4.

CentOS installer runs ansible installer and installs some stuff with yum then exits when it's done, nothing else happens. Running the script again fails with `fatal: destination path 'ansible' already exists and is not an empty directory.`.

Installing on Ubuntu actually runs the installer script but a lot of things are missing. The devshop bin executible is not found, going to the app requires composer install be run because it hadn't been run yet. Composer is not even on the machine (tried running with composer installed). Drush also does not get installed. If I try to set the webserver to nginx, it still installs apache. Setting the hostname flag does not seem to have any effect on the server's hostname.

I thought this thing was supposed to provision the server? Am I missing something here?

**Comment by jonpugh** [comment]
Hi Adam,
I am going to merge changes to our CI in #575 today which will clear the way for clearing up install issues, along with fixing the tests so `install.sh` is properly covered.

I'm not sure about most of your problems right this second. We run automated tests of the playbooks on GitHub actions, but they are built on the @geerlingguy Ansible containers.

If you look at his Docker file: https://github.com/geerlingguy/docker-ubuntu1804-ansible/blob/master/Dockerfile

Run those commands before trying install.sh, and let me know how it goes?

Stand by for improved install testing PR later today.

**Comment by asentner** [comment]
@jonpugh Thanks for the update. I'll see if I can find some time tonight to try out your idea. I'll report back as soon as I can!

Thanks a bunch!!

**Comment by mazzarito** [comment]
I was able to get it to install recently by using —debug and —verbose options on install.sh and resolving each issue as it arose sometimes running the commands manually

**Comment by asentner** [comment]
@jonpugh Still no luck with the installer. Ran all the commands beforehand in the dockerfile with the exception of removing the udev devices. It failed on the MySQL install:

```
TASK [geerlingguy.mysql : Remove anonymous MySQL users.] **************************************************************************************************************

TASK [geerlingguy.mysql : Remove MySQL test database.] ****************************************************************************************************************
fatal: [devshop.sprowt.us]: FAILED! => changed=false 
  msg: The PyMySQL (Python 2.7 and Python 3.X) or MySQL-python (Python 2.X) module is required.

RUNNING HANDLER [geerlingguy.apache : restart apache] *****************************************************************************************************************

RUNNING HANDLER [geerlingguy.php : restart webserver] *****************************************************************************************************************

RUNNING HANDLER [geerlingguy.mysql : restart mysql] *******************************************************************************************************************

PLAY RECAP ************************************************************************************************************************************************************
devshop.sprowt.us          : ok=104  changed=39   unreachable=0    failed=1    skipped=51   rescued=0    ignored=0   
```

I'm guessing it does not finish running? Still notice drush, composer, devshop is not in the /usr/local/bin or other bin directories. Does not appear to be a complete installation.

I'll try to get MySQL installed first before running the script but this is something it should probably be able to handle right?

I really want to get this to work so any feedback would definitely be appreciated.

**Comment by asentner** [comment]
@jonpugh Still no luck with the installer. Ran all the commands beforehand in the dockerfile with the exception of removing the udev devices. It failed on the MySQL install:
```
TASK [geerlingguy.mysql : Remove anonymous MySQL users.] **************************************************************************************************************

TASK [geerlingguy.mysql : Remove MySQL test database.] ****************************************************************************************************************
fatal: [devshop.example.com]: FAILED! => changed=false 
  msg: The PyMySQL (Python 2.7 and Python 3.X) or MySQL-python (Python 2.X) module is required.

RUNNING HANDLER [geerlingguy.apache : restart apache] *****************************************************************************************************************

RUNNING HANDLER [geerlingguy.php : restart webserver] *****************************************************************************************************************

RUNNING HANDLER [geerlingguy.mysql : restart mysql] *******************************************************************************************************************

PLAY RECAP ************************************************************************************************************************************************************
devshop.example.com         : ok=104  changed=39   unreachable=0    failed=1    skipped=51   rescued=0    ignored=0   
```

I'm guessing it does not finish running? Still notice drush, composer, devshop is not in the /usr/local/bin or other bin directories. Does not appear to be a complete installation.

I'll try to get MySQL installed first and PyMySQL before running the script but this is something it should probably be able to handle right?

I really want to get this to work so any feedback would definitely be appreciated.

@mazzarito Do you remember what errors you were facing? I am running Ubuntu 18.4.4 LTS (minimal installation), how about you?

**Comment by mazzarito** [comment]
@asentner Oh I remember that one, just install pymysql: 

`sudo apt install python-pymysql` or `sudo apt install python3-pymysql`

I forget which version its wanting at that point, so you could just do both.

**Comment by asentner** [comment]
@mazzarito That's what I ended up doing to get it to finish.

@jonpugh These are the commands I ran before the installer to get a successful install:

```
wget https://raw.githubusercontent.com/opendevshop/devshop/1.x/install.sh \
  -O install-devshop.sh
chmod +x install-devshop.sh 
apt update
apt upgrade
apt install -y --no-install-recommends apt-utils locales python3-setuptools \
  python3-pip software-properties-common rsyslog systemd systemd-cron \
  sudo iproute2 python python-setuptools python-pip python-pymysql \
  python3-pymysql
sed -i 's/^\($ModLoad imklog\)/#\1/' /etc/rsyslog.conf
pip3 install ansible
mkdir -p /etc/ansible
echo "[local]\nlocalhost ansible_connection=local" > /etc/ansible/hosts
./install-devshop.sh 
```

Hope that helps in fixing the issues with the installer on Ubuntu 18.4.4 LTS

**Comment by jonpugh** [comment]
For reference, This is exactly what happens in the prepared geerlingguy Docker containers. Did you know? https://github.com/geerlingguy/docker-ubuntu1804-ansible/blob/master/Dockerfile

Using the geerlingguy containers is why we didn't catch this is automated testing.

Stand by for fixes coming in #580: the install.sh script now installs PIP, then uses that to install ansible.

Thanks for confirming the missing python-mysql packages, @mazzarito ... I decided to add it to the "pip_packages" in `install.sh` ... however, I may be able to remove it because it could work now I am going back to forcing python3 using `ansible_python_interpreter`...

Thank you all for this triaging!

**Comment by jonpugh** [comment]
Really it all came down to this:

https://github.com/opendevshop/devshop/pull/580/commits/7ad2e52bb132945244701b12b3b949f26e27b700

💯
